planner/core/find_best_task.go

```diff
@@ -1677,6 +1692,7 @@ func (ds *DataSource) convertToBatchPointGet(prop *property.PhysicalProperty, ca
 		batchPointGetPlan.IndexInfo = candidate.path.Index
 		batchPointGetPlan.IdxCols = candidate.path.IdxCols
 		batchPointGetPlan.IdxColLens = candidate.path.IdxColLens
+		batchPointGetPlan.PartitionColPos = getPartitionColumnPos(candidate.path.Index, hashPartColName)
```

Comment:
What if `hashPartColName` is not in this index? I see `getPartitionColumnPos` will cause a panic in that case, but I don't know whether this can happen here.

Reply:
Hash partition column must be in every unique index, according to https://docs.pingcap.com/zh/tidb/stable/partitioned-table#%E5%88%86%E5%8C%BA%E7%9A%84%E7%BA%A6%E6%9D%9F%E5%92%8C%E9%99%90%E5%88%B6